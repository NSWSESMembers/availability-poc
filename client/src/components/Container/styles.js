import { StyleSheet } from 'react-native';
import Colors from '../../themes/Colors';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.bgMain,
  },
  holder: {
    padding: 25,
  },
});

export default styles;
