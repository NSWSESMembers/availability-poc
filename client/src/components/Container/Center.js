import PropTypes from 'prop-types';
import React from 'react';
import { View } from 'react-native';
import styles from './styles';

const Center = ({ children }) => <View style={styles.containerCenter}>{children}</View>;

Center.propTypes = {
  // eslint-disable-next-line
  children: PropTypes.any,
};

export default Center;
