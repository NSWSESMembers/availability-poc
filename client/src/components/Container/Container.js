import PropTypes from 'prop-types';
import React from 'react';
import { View } from 'react-native';
import styles from './styles';

const Container = ({ children, isAlt }) => {
  let containerStyles = [styles.container];

  if (isAlt === true) {
    containerStyles = [styles.containerAlt];
  }

  return <View style={containerStyles}>{children}</View>;
};

Container.propTypes = {
  // eslint-disable-next-line
  children: PropTypes.any,
  isAlt: PropTypes.bool,
};

Container.defaultProps = {
  isAlt: false,
};

export default Container;
