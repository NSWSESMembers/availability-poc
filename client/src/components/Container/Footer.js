import PropTypes from 'prop-types';
import React from 'react';
import { View } from 'react-native';
import styles from './styles';

const Footer = ({ children }) => (
  <View style={styles.containerFooter}>
    {children}
  </View>
);

Footer.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Footer;
