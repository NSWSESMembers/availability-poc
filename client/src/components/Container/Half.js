import PropTypes from 'prop-types';
import React from 'react';
import { View } from 'react-native';
import styles from './styles';

const Half = ({ children }) => <View style={styles.containerHalf}>{children}</View>;

Half.propTypes = {
  // eslint-disable-next-line
  children: PropTypes.any,
};

export default Half;
