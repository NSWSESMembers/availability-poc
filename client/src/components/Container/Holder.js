import PropTypes from 'prop-types';
import React from 'react';
import { View } from 'react-native';
import styles from './styles';

const Container = ({
  children,
  margin,
  marginVertical,
  marginTop,
  paddingVertical,
  wide,
  marginBot,
  transparent,
}) => {
  const containerStyles = [styles.containerHolder];

  if (paddingVertical) {
    containerStyles.push({ paddingTop: 8, paddingBottom: 8 });
  }

  if (marginVertical) {
    containerStyles.push({ marginTop: 8, marginBottom: 8 });
  }

  if (marginTop) {
    containerStyles.push({ marginTop: 8 });
  }

  if (margin) {
    containerStyles.push({ paddingTop: 20 });
  }

  if (wide) {
    containerStyles.push({ paddingLeft: 5, paddingRight: 5 });
  }

  if (marginBot) {
    containerStyles.push({ paddingBottom: 20 });
  }
  if (transparent) {
    containerStyles.push({ backgroundColor: 'transparent' });
  }

  return <View style={containerStyles}>{children}</View>;
};

Container.propTypes = {
  children: PropTypes.node.isRequired,
  paddingVertical: PropTypes.bool,
  marginTop: PropTypes.bool,
  marginVertical: PropTypes.bool,
  margin: PropTypes.bool,
  marginBot: PropTypes.bool,
  wide: PropTypes.bool,
  transparent: PropTypes.bool,
};

export default Container;
