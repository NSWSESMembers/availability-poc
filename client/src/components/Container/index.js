import Center from './Center';
import Container from './Container';
import Footer from './Footer';
import Half from './Half';
import Holder from './Holder';
import styles from './styles';

export { Center, Container, Footer, Half, Holder, styles };
