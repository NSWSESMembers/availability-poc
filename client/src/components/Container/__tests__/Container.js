import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import Container from '../Container';

test('renders correctly', () => {
  const tree1 = renderer.create(
    <Container>
      <Text>test</Text>
    </Container>,
  ).toJSON();
  expect(tree1).toMatchSnapshot();

  const tree2 = renderer.create(
    <Container isAlt>
      <Text>test</Text>
    </Container>,
  ).toJSON();
  expect(tree2).toMatchSnapshot();
});
