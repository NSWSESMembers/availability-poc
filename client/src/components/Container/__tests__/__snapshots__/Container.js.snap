// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#EEE",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    test
  </Text>
</View>
`;

exports[`renders correctly 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#1e3c78",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    test
  </Text>
</View>
`;
